src/main/java/com/pusher/rest/data/Result.java

```diff
@@ -32,11 +32,11 @@ public boolean shouldRetry() {
         }
     }
 
-    private final Status status;
-    private final Integer httpStatus;
-    private final String message;
+    protected final Status status;
+    protected final Integer httpStatus;
+    protected final String message;
```

Comment:
I would leave these three private and use the accessor methods in the sub-class. There's no reason for the subclass to have direct access to these read-only properties.